### PR TITLE
`ParamSpec` must not raise errors on valid attr access

### PR DESCRIPTION
**Pull request by sobolevn**
Refs https://github.com/python/mypy/pull/13468

CC @JelleZijlstra

### CODE REVIEW

File: test-data/unit/check-parameter-specification.test [hunk 1]
```diff
@@ -1127,3 +1127,38 @@ def func(callback: Callable[P1, str]) -> Callable[P1, str]:
...
+[case testParamSpecTestPropAccess]
+from typing import Callable
+from typing_extensions import ParamSpec
+
+P1 = ParamSpec("P1")
+
+def func1(callback: Callable[P1, str]) -> Callable[P1, str]:
+    def inner(
+        *args: P1.typo,  # E: Use "P1.args" for variadic "*" parameter \
+                         # E: Name "P1.typo" is not defined
```
Comment (JelleZijlstra) [inline test-data/unit/check-parameter-specification.test:1140]: How hard would it be to make this say something like "ParamSpec has no attribute 'typo'" instead?

**Review by JelleZijlstra — COMMENTED** [review]
Thanks!

File: test-data/unit/check-parameter-specification.test (hunk 1 above)
Comment (sobolevn) [reply test-data/unit/check-parameter-specification.test:1140]: Quite hard :)

Right now `checkmember` is not involved at all, this is `semanal`-level check.
We can change the error message (special case it for `ParamSpec`), but I don't think this is a good idea.

**Review by JelleZijlstra — APPROVED** [review]

**Comment by sobolevn** [comment]
Thanks for noticing it!

**Comment by vnmabus** [comment]
This still fails for me in the following example in Mypy playground (https://mypy-play.net/?mypy=master&python=3.10&gist=2f63e549be8913d2878f366612bb7b00):

```python
from typing import ParamSpec, Callable, TypeVar

T = TypeVar("T")
P = ParamSpec("P")

def twice(
    f: Callable[P, T],
    **kwargs: P.kwargs,
) -> T:
  return f(**kwargs)
```

Is this not supposed to work?

**Comment by hauntsaninja** [comment]
It works just fine on master, the problem is that mypy-play.net's master is just some really old hardcoded revision. See https://github.com/ymyzk/mypy-playground/issues/162 / https://github.com/ymyzk/mypy-playground/issues/158

**Comment by vnmabus** [comment]
> It works just fine on master, the problem is that mypy-play.net's master is just some really old hardcoded revision. See [ymyzk/mypy-playground#162](https://github.com/ymyzk/mypy-playground/issues/162) / [ymyzk/mypy-playground#158](https://github.com/ymyzk/mypy-playground/issues/158)

Ok, I assumed it was automatically pulling the latest changes 😅️.

**Pull request merged**